eng/generators.targets

```diff
@@ -63,8 +63,28 @@
                             and @(ProjectReference->AnyHaveMetadataValue('Identity', '$(CoreLibProject)'))))" Include="$(LibrariesProjectRoot)Common\src\System\Runtime\InteropServices\ArrayMarshaller.cs" />
   </ItemGroup>
 
+  <ItemGroup>
+    <EnabledGenerators Include="RegexGenerator" Condition="'$(EnableRegexGenerator)' == 'true'" />
+    <!-- Enable the RegexGenerator source generator when the project is a C# source project that references System.Text.RegularExpressions -->
+    <EnabledGenerators Include="RegexGenerator"
```

Comment:
Based on previous feedback by me and @ericstj, I believe we want to remove this implicit entrypoint and leave it to just the one above that depends on the property.